CloudEvents Project Security Self-Assessment - Security Pals

Igor8mr opened the pull request:
Created and added the first draft of the CloudEvents Project Security Self-Assessment.

Please feel free to share your thoughts on the security self-assessment.

eddie-knight commented:
Hi there, and thanks for the hard work you put into creating this self-assessment!

We have plenty more to review on the PR, but as a starter I noticed that you have some notes written in `assessments/projects/cloudevents/CE-maintainers-communications.md`. Could you move these notes to the PR description, comments, or remove them (as needed)?

eddie-knight commented on assessments/projects/cloudevents/files/CloudEvents SBOM.spdx:
There are two reasons that jump to the front of my mind for why an SBOM isn't needed in this PR...

1. SBOMs should be associated with releases, as the bill of materials is only accurate and useful if it is created at build time and associated to a particular point in the code history.
1. If you need to link to an SBOM for some reason in the self-assessment, you can just provide a link out to the latest build artifacts that contain an SBOM.

We still have plenty more to review, but as a starter— could you please remove the SBOM from this PR?

Igor8mr replied:
Hi @eddie-knight, thank you for your comments!
That makes sense, I just removed the notes and the SBOM.
Please, let me know if you have more comments in the future. Thank you!

Igor8mr commented:
Below are the communications we had with the CloudEvents Security Team to write this Security Assessment.

## Slack Communications

* **Security Pals Involved:**
  * Igor Rodrigues (Igor Rodrigues)
* **CloudEvents Team Members Involved:**
  * Doug Davis (dug)

### Slack Report

#### Igor Rodrigues (Nov 29th at 4:29:13 PM)

* [Message
  Link](https://cloud-native.slack.com/archives/C9DB5ABAA/p1701293353624819)

Hello all, I'm a student at NYU involved in the SecurityPal effort from TAG
Security. Our group is conducting a security assessment on CloudEvents, which we
will later submit to the [TAG Security Assessments
Repository](https://github.com/cncf/tag-security/tree/main). We have completed
an [initial
evaluation](https://github.com/Igor8mr/tag-security/blob/main/assessments/projects/cloud-events/self-assessment.md)
of the project and would appreciate your feedback to validate the information we
included. We also want to know if there are additional aspects we should include
in the assessment to correctly represent your project, along with more details
for sections like [security issue
resolution](https://github.com/Igor8mr/tag-security/blob/main/assessments/projects/cloud-events/self-assessment.md#security-issue-resolution)
and [secure development
practices](https://github.com/Igor8mr/tag-security/blob/main/assessments/projects/cloud-events/self-assessment.md#secure-development-practices).
Please, feel free to share your thoughts here on Slack, on GitHub, or on a call.
Thank you!

#### Dug (Nov 29th at 8:02:04 PM)

* [Message
  Link](https://cloud-native.slack.com/archives/C9DB5ABAA/p1701306124114029?thread_ts=1701293353.624819&cid=C9DB5ABAA)

Hi @Igor Rodrigues - will take a look. Just curious though, what made you decide
to analyze CloudEvents?

#### Igor Rodrigues (Nov 29th at 8:36:26 PM)

* [Message
  Link](https://cloud-native.slack.com/archives/C9DB5ABAA/p1701308186825319?thread_ts=1701293353.624819&cid=C9DB5ABAA)

Hi @dug, thank you. The assessment is one of our assignments for a class we are
taking with Professor Justin Cappos. Each group was assigned to a CNCF project,
and ours was CloudEvents. The project is interesting, so we are trying to do a
bit more than expected. I hope the assessment helps in the future.

#### Dug (Nov 30th at 10:24:13 AM)

* [Message
  Link](https://cloud-native.slack.com/archives/C9DB5ABAA/p1701357853677559?thread_ts=1701293353.624819&cid=C9DB5ABAA)

@Igor Rodrigues thanks. Just a few comments from my quick scan:

* Where do you see ANTRL being used? I'm surprised you didn't include markdown
  in the list despite it not being a "programming language", being a "spec"
  markdown is kind of our "language" :slightly_smiling_face:
* `CloudEvents was developed to address the lack of uniformity in event data
  format...` be a bit careful here. While CE does provide a "structured" format,
  that's just there are times when people want the event data and context
  attributes in one doc. In general though CE is NOT trying to define "yet
  another common event format (one format to rule them all)". In particular,
  many people use/prefer "binary" format because it just augments their existing
  events. And even with "structured", the stuff that does into the `data`
  attribute is wide open - and should be defined by the business. I just don't
  want people to think we're making the same mistake as other folks who tried to
  force one format for all events. Rather CE is about standardizing "where to
  find common metadata about the event w/o having to parse/understand the event
  specific format".
* Nit: in "Protocol Binding" section it mentions `structured-mode` but hasn't
  defined that term yet. You may want to define binary vs structured CEs in the
  doc before this section.
* Not sure what the "trust boundary" is meant to represent in the diagram since
  "trust" is kind of orthogonal to the roles.
* Goals: may want to tweak some of those based on my comments above. Plus, some
  of those aren't really goals for CE since CE doesn't control them. For
  example, "generate events before consumers are listening" - a good idea, but
  CE doesn't really talk about those in the spec itself. CE is just about the
  format and how they might appear on the transports. With a few exceptions, it
  doesn't get into the protocols themselves or event
  management/subscriptions.....
* CE is under review for Graduation status right now... hopefully will be
  approved very soon
* CE doesn't really describe any encryption mechanism or deal with integrity -
  the text you wrote kind of implies CE addresses it. Perhaps say something like
  it's an implementation detail/choice??
* Ecosystem - might be good to link to the [cloudevents.io](cloudevents.io) site
  which includes a list of adopters.
* The "Security issue resolution" section reads like an SDK specific section -
  perhaps "SDK" should appear in the title to make it clear that the following
  sections apply to the SDK repos and not the spec repo?
* There's also a new security mailing list people should use to report security
  concerns: https://lists.cncf.io/g/cncf-cloudevents-security/topics
* There is no "CloudEvents Steering Committee" that's mentioned in the Threat
  Modelling section (typo in Modelling)
* It might be good to mention that (I think) all of the security issues found by
  Trail of Bits have been addressed

#### Igor Rodrigues (Nov 30th at 11:58:51 AM)

* [Message
  Link](https://cloud-native.slack.com/archives/C9DB5ABAA/p1701363531073659?thread_ts=1701293353.624819&cid=C9DB5ABAA)

Hi @dug, Thank you for all the comments! For ANTLR, GitHub marked it as 14.1% of
the [CloudEvents spec](https://github.com/cloudevents/spec), so that's why I
added it to the assessment, but I may remove it if it's not very relevant. I'll
also definitely add Markdown, thanks for noticing that. We'll review the doc,
update it with your comments and tell you about the changes. Thank you again!

#### Igor Rodrigues (Dec 4th at 11:15:26 AM)

* [Message
  Link](https://cloud-native.slack.com/archives/C9DB5ABAA/p1701706526314599?thread_ts=1701293353.624819&cid=C9DB5ABAA)

Hi @dug, we fixed the comments you provided on the [security
assessment](https://github.com/Igor8mr/tag-security/blob/main/assessments/projects/cloud-events/self-assessment.md),
along with the comments from the meeting. Here are the [new
changes](https://github.com/cncf/tag-security/commit/e75e0e0a908ffa462c7923fad6e6e201b5feaef0#diff-086780f8339d58b8abcf32f9cf930f8b11ebf1889ee3e36c4eeaede7dc21a7b7)
since then. Please, let me know if there are more parts we could improve. Also,
I wanted to CloudEvents have a public SBOM that we could link, and if you think
there are more aspects we could add to the specification side of the [Security
Issue
resolution](https://github.com/Igor8mr/tag-security/blob/main/assessments/projects/cloud-events/self-assessment.md#cloudevents-specification).
Thank you for all the help!

#### Dug (Dec 4th at 11:36:23 AM)

* [Message
  Link](https://cloud-native.slack.com/archives/C9DB5ABAA/p1701707783421699?thread_ts=1701293353.624819&cid=C9DB5ABAA)

The closest thing we have to a SBOM is:
https://github.com/cloudevents/spec#cloudevents-documents Thanks for the update.
Will look it over in a bit.

#### Igor Rodrigues (Dec 4th at 11:44:52 AM)

* [Message
  Link](https://cloud-native.slack.com/archives/C9DB5ABAA/p1701708292972649?thread_ts=1701293353.624819&cid=C9DB5ABAA)

Great, thanks!

#### Dug (Dec 4th at 12:08:22 PM)

* [Message
  Link](https://cloud-native.slack.com/archives/C9DB5ABAA/p1701709702994029?thread_ts=1701293353.624819&cid=C9DB5ABAA)

I put just a few minor tweaks as comments on the commit.

#### Igor Rodrigues (Dec 4th at 12:28:53 PM)

* [Message
  Link](https://cloud-native.slack.com/archives/C9DB5ABAA/p1701710933601919?thread_ts=1701293353.624819&cid=C9DB5ABAA)

Thanks, I'll fix those soon

#### Igor Rodrigues (Dec 5th at 8:05:09 AM)

* [Message
  Link](https://cloud-native.slack.com/archives/C9DB5ABAA/p1701781509377939?thread_ts=1701293353.624819&cid=C9DB5ABAA)

Hi @dug, I forgot to ask this before, but are there any action items you are
currently working on or plan to work on that would solve the concerns mentioned
in the doc or other security concerns? I think it would be good to include those
in the assessment. I remember you mentioned implementing bots to check the SDKs,
do you have a pull request, issue, or any other link to the implementation of
the bots idea? Also, we are willing to help implement one of those solutions to
the concerns if you have some specific things in mind.

#### Dug (Dec 5th at 11:57:30 AM)

* [Message
  Link](https://cloud-native.slack.com/archives/C9DB5ABAA/p1701795450643219?thread_ts=1701293353.624819&cid=C9DB5ABAA)

@Igor Rodrigues just this one: https://github.com/cloudevents/spec/issues/1235

#### Dug (Dec 5th at 11:58:19 AM)

* [Message
  Link](https://cloud-native.slack.com/archives/C9DB5ABAA/p1701795499076589?thread_ts=1701293353.624819&cid=C9DB5ABAA)

If someone knows how to setup the bots and wants to submit a PR to add them...
that would be great! Or even just a list of instructions for an admin to follow
(if it's more than just a PR) that would be great too.

#### Igor Rodrigues (Dec 5th at 12:12:57 PM)

* [Message
  Link](https://cloud-native.slack.com/archives/C9DB5ABAA/p1701796377013619?thread_ts=1701293353.624819&cid=C9DB5ABAA)

Great, thanks! We are taking a look here

## CloudEvents Team Meeting

* **Security Pals Involved:**
  * Igor Rodrigues
* **CloudEvents Team Members Involved:**
  * Doug Davis
  * Tommy
  * Erik
  * David B
  * Jon
  * Calum
  * Jem
  * Clemens

### Team Meeting Report

The team joined the CloudEvents public team meeting on November 30th, 2023,
which was [recorded on
YouTube](https://www.youtube.com/watch?v=2OZPTQOqFEw&t=191s).

duglin commented:
Can we please not call this a "self-assessment" since this was not done by the CE project itself. I'm not suggesting anything is incorrect with the assessment but it's not accurate to imply that the CE team did it. This was done as a school project (at the request of their teacher) and while the CE team did look it over, it was done so just to make sure there weren't any superficial problems with it. If this is meant to be some kind of formal CNCF review/analysis then more work/review should be done so it can have a more official standing.

For the TAG-Security team... what's the purpose of these docs? What are they used for? Do they have any official standing within the CNCF? I've looked over https://github.com/cncf/tag-security/tree/main/assessments but what's not clear to me is why a project should create one of these docs if they've already done something like had a Trail of Bits review and added that analysis to their own docs.

JustinCappos commented:
> Can we please not call this a "self-assessment" since this was not done by the CE project itself.

STAG Security Assessment Facilitator here.. I understand the potential for confusion.  The name "self-assessment" was created back when we made the original process because it was actually done by the projects.  The Security Pals process was added because some CNCF projects hadn't done their self assessments and these Pals are meant to help spur it along.  I totally get where you're coming from, we will certainly consider this feedback.


>  I'm not suggesting anything is incorrect with the assessment but it's not accurate to imply that the CE team did it. This was done as a school project (at the request of their teacher) and while the CE team did look it over, it was done so just to make sure there weren't any superficial problems with it. If this is meant to be some kind of formal CNCF review/analysis then more work/review should be done so it can have a more official standing.

Yes, please do this!  This is the goal!  

> For the TAG-Security team... what's the purpose of these docs? What are they used for? Do they have any official standing within the CNCF? I've looked over https://github.com/cncf/tag-security/tree/main/assessments but what's not clear to me is why a project should create one of these docs if they've already done something like had a Trail of Bits review and added that analysis to their own docs.

So they are part of the assessment process in the TAG Security repo ( https://github.com/cncf/tag-security/tree/main/assessments ).  At different times the TOC has required / recommended these for projects.  We have heard from end user adopters, etc. that they find these valuable in getting another perspective on a project.  This is different than an audit from ToB, etc. which tends to be a more "point-in-time" security assessment of specific components of a project.  

I'd be happy to discuss any of this process, etc. further either in a TAG Security meeting (I am usually on the Wednesday call), on the CNCF slack, or via a separate issue dedicated to revising the assessment process, etc. on the TAG Security issue tracker.

Igor8mr commented:
Thank you for your comments explaining the purpose of the project @JustinCappos! I would also be happy to discuss the assessment with you and the CloudEvents team on a call, on Slack, or here. 

@duglin, please, feel free to add more comments here or on Slack with more suggestions for the assessment if you can. We want to represent the project in the most accurate way possible, so we are definitely open to your feedback. Thank you again for all the feedback you have given me so far.

Rana-KV commented on assessments/projects/cloudevents/self-assessment.md:
```diff
@@ -0,0 +1,1051 @@
+
+## Project compliance
+
+As of the latest security assessment, CloudEvents does not explicitly document
+compliance with specific security standards such as PCI-DSS, COBIT, ISO, GDPR,
+etc. Current efforts are focused on evaluating compliance with these standards
+and ensuring that CloudEvents adheres to industry best practices in security and
+privacy.
```
Thanks for the documentation, can you remove the 2nd sentence as it can be misconstrued.

Igor8mr replied:
Thank you for reviewing the assessment. Sounds good, I removed the sentence now.

Rana-KV commented on assessments/projects/cloudevents/self-assessment.md:
```diff
@@ -0,0 +1,1051 @@
+
+### CloudEvents SDKs
+
+#### Responsible Disclosures Process
```
Provide links to the Responsible Disclosures Process and Vulnerability Response Process documented in CloudEvents. Additionally, update the content in more details.

Igor8mr replied:
Thank you for the comment! I added new links to documents CloudEvents has that mention these processes. Unfortunately, they don't have a document outlining these processes in further detail, so I also added a new action item on `Improve the documentation on the Security Response Processes`. I also messaged the CloudEvents team now to ask if they have more details on these processes that I could add to the assessment. I will update it further when they give me more information. Please, let me know if I should make more changes. Thank you!

Rana-KV commented on assessments/projects/cloudevents/self-assessment.md:
Add explicitly that the project doesn't have one CII best practices badge. You also consider one of the action items to improve alignment with CII best practices and to get one.

Igor8mr replied:
Thank you for the comment. I found that CloudEvents received a `passing` [badge in 2022](https://www.bestpractices.dev/en/projects/6770), so I just added it to the assessment now. I also added a new action item on `Obtain a Silver or Gold CII Best Practices Badge`. Please, let me know if this is good or if I should make more changes!

Rana-KV replied:
Yup, the changes good.

ragashreeshekar commented on assessments/projects/cloudevents/self-assessment.md:
```diff
@@ -0,0 +1,1051 @@
+| Security Provider | CloudEvents is not a security provider  |
+| CloudEvents Specification Language | Markdown |
+| CloudEvents SDK Languages | Python, C#/.NET, Go, Java, Javascript, PHP, PowerShell, Ruby and Rust |
+| Technical Documents | [CloudEvents Documents](https://github.com/cloudevents/spec#cloudevents-documents) |
```
Please add link to SBOM if the project provides one

Igor8mr replied:
Thank you for reviewing the assessment! Unfortunately, CloudEvents does not have a public SBOM, but I added `CloudEvents does not have a public SBOM available` on the table now. Please, let me know if this is fine or if I should make more changes! 

We have also started working on your other comments and will bring more updates soon. Thank you!

ragashreeshekar commented on assessments/projects/cloudevents/self-assessment.md:
```diff
@@ -0,0 +1,1051 @@
+
+#### The SDK management teams may implement vulnerabilities while implementing SDKs
+
+Trail Of Bits was able to identify 7 different security concerns regarding the
```
This seems to be the work outside of the scope of security pals activity. I would recommend introducing it in a few sentences and providing its reference instead of copying the same content here.

Igor8mr replied:
Thank you for the feedback. I shortened this sentence and added a reference link there. Please, let me know if I should make more changes there.

ragashreeshekar commented on assessments/projects/cloudevents/self-assessment.md:
This title is not relevant to the content, as it implies project needs to implement enhanced encryption and data validation mechanisms. Something on the lines of document project scope and user responsibilities might a better fit.

Igor8mr replied:
Thank you for noticing that, I changed the title to `Emphasize goals, non-goals and user responsibilities`.

ragashreeshekar commented on assessments/projects/cloudevents/self-assessment.md:
Kindly keep the recommendations to reflect the task instead of suggesting particular tool. In this case, the task is to establish SAST, SCA and/or secret scanning services. The project team is free to choose the tool of their choice to accomplish this. 

In addition, pointer to associated task/process documentation could be useful, however I would suggest removing the steps to accomplish it embedded in this document.

Igor8mr replied:
Thank you for the comment! I changed the comment to make it broader now. I also linked this to a GitHub issue with the task/process documentation and removed the steps from the documentation.

ragashreeshekar commented on assessments/projects/cloudevents/self-assessment.md:
This seems to be the work outside of the scope of security pals activity. I would recommend introducing it in a few sentences and providing its reference instead of copying the same content here.

Igor8mr replied:
Thank you for the feedback. We removed this part and replaced it with a summary and reference.

ragashreeshekar commented on assessments/projects/cloudevents/self-assessment.md:
This section is intended to introduce related projects, and state how it is different from the current project in scope. I see the former, however the latter is absent. Please add a few sentences on how these projects are different from CloudEvents.

Igor8mr replied:
Thank you for the feedback. I added a paragraph under each related project to compare them with CloudEvents. Please, let me know if I should add more details there.

ragashreeshekar commented on assessments/projects/cloudevents/self-assessment.md:
```diff
@@ -0,0 +1,1051 @@
+is the entity or system that is responsible for the distribution, processing and
+routing of events to consumers. The event broker ensures reliable delivery and
+may enforce security policies.
+
```
Does this assessment cover the SDKs offered by CloudEvents team or the scope is limited to the spec? 
If its the former, is the actor/actions updated according to the implementation?

Igor8mr replied:
Thank you for noticing that. It covers both the SDK and the Specification, so we added more actors now to cover the SDK part. Please, let me know if I should make more changes.

ragashreeshekar reviewed (changes requested):
Thanks for the PR @Igor8mr and team, appreciate the efforts.
I have completed first pass of review. Please feel free to reach out here or on slack for any questions and clarifications.

Along with addressing the comments, kindly update the PR branch with the latest content in the repo as this branch is out-of-date with the base branch.

duglin commented:
> So they are part of the assessment process in the TAG Security repo ( https://github.com/cncf/tag-security/tree/main/assessments ). At different times the TOC has required / recommended these for projects. We have heard from end user adopters, etc. that they find these valuable in getting another perspective on a project. This is different than an audit from ToB, etc. which tends to be a more "point-in-time" security assessment of specific components of a project.

Thanks but I'm not sure this really answers my question. TOB did a review from a security perspective and we include that in our repo. This PR doesn't introduce much (any?) new 'security' related information and actually spends most of its time talking about the project from a process or governance perspective. Not bad things but it feels a bit odd to be part of a "security" review document. Yes the TOB review is a point in time review, but so is this PR... or any assessment. I'm just trying to understand the impact/benefit of this piece of work when it feels like just about all of the info is already part of our repo (or should be as that's the source of truth for all things CE) so why duplicate it here, and are we then on the hook to keep it up to date since the minute the PR is merged it could be out of date?

Sorry if this sounds a bit like I'm pushing back, but I have a very strong negative reaction to things that feel like unnecessary bureaucracy - especially if it then requires an on-going commitment.

eddie-knight commented on assessments/projects/cloudevents/self-assessment.md:
```diff
@@ -0,0 +1,1063 @@
+Contributors/Reviewers: Pranava Kumar Vemula (@Rana-KV), Doug Davis (@duglin),
+Eddie Knight (@eddie-knight) and Ragashree (@ragashreeshekar).
+
+This document elaborates and explores the design goals for
+[CloudEvents](https://cloudevents.io/) as well as a security assessment of the
+software. It was written in collaboration with the CloudEvents team. All
+communications with CloudEvents were recorded on a [separate
+document](CE-maintainers-communications.md).
```
I like that you put this here! But since we removed the comms file, we should restructure the paragraph.

```suggestion
This document was written in collaboration with the CloudEvents team to elaborate on and explores the design goals for
[CloudEvents](https://cloudevents.io/) through a limited security assessment of the software.
```

Igor8mr replied:
Thank you, I committed the suggestion.

eddie-knight commented on assessments/projects/cloudevents/self-assessment.md:
```diff
@@ -0,0 +1,1063 @@
+  * [Potential threats](#potential-threats)
+  * [Conclusion](#conclusion)
+  * [Recommendations](#recommendations)
+* [Action Items](#action-items)
+  * [[Action Item 1] Setup a system for automatically reviewing
+    SDKs](#action-item-1-setup-a-system-for-automatically-reviewing-sdks)
+  * [[Action Item 2] Regular SDK Security
+    Audits](#action-item-2-regular-sdk-security-audits)
+  * [[Action Item 3] Enforce governance rules throughout SDK
+    development](#action-item-3-enforce-governance-rules-throughout-sdk-development)
+  * [[Action Item 4] Obtain a Silver or Gold CII Best Practices
+    Badge](#action-item-4-obtain-a-silver-or-gold-cii-best-practices-badge)
+  * [[Action Item 5] Improve the documentation on the Security Response
+    Processes](#action-item-5-improve-the-documentation-on-the-security-response-processes)
+  * [[Action Item 6] Emphasize goals, non-goals and user
+    responsibilities](#action-item-6-emphasize-goals-non-goals-and-user-responsibilities)
+  * [[Action Item 7] Documentation and Knowledge
+    Sharing](#action-item-7-documentation-and-knowledge-sharing)
+  * [[Action Item 8] Community Engagement and
+    Feedback](#action-item-8-community-engagement-and-feedback)
```
Following discussions from the review team, we are asking that Security Pals please remove any recommendations from the scope of this document (unless they were written and included under guidance from the maintainer team). You may optionally communicate Security Pals recommendations for projects in a more accessible location, such as in a project community channel or GitHub Issue.

Igor8mr replied:
Hi @eddie-knight, thank you for the feedback on this. I removed all the action items, except the `Setup a system for automatically reviewing SDKs`, which is something we discussed with the maintainer team, as you mentioned not to remove those. They are also already [working on implementing this](https://github.com/cloudevents/spec/issues/1235) and we provided some [guidance to help them get started](https://github.com/cloudevents/spec/issues/1235#issuecomment-1841532959).  I also moved this action item to the appendix now as you mentioned.

Please, let me know if I should still change this section more, or remove this last action item. Thank you again!

Igor8mr commented on assessments/projects/cloudevents/self-assessment.md:
```diff
@@ -0,0 +1,1063 @@
+| Security Provider | CloudEvents is not a security provider  |
+| CloudEvents Specification Language | Markdown |
+| CloudEvents SDK Languages | Python, C#/.NET, Go, Java, Javascript, PHP, PowerShell, Ruby and Rust |
+| SBOM | CloudEvents does not have a public SBOM available |
```
Thank you for the suggestion! I committed it.

Igor8mr commented on assessments/projects/cloudevents/self-assessment.md:
Thank you for your comment, I merged it to the appendix now.

JustinCappos commented:
> > So they are part of the assessment process in the TAG Security repo ( https://github.com/cncf/tag-security/tree/main/assessments ). At different times the TOC has required / recommended these for projects. We have heard from end user adopters, etc. that they find these valuable in getting another perspective on a project. This is different than an audit from ToB, etc. which tends to be a more "point-in-time" security assessment of specific components of a project.
> 
> Thanks but I'm not sure this really answers my question. TOB did a review from a security perspective and we include that in our repo. This PR doesn't introduce much (any?) new 'security' related information and actually spends most of its time talking about the project from a process or governance perspective. Not bad things but it feels a bit odd to be part of a "security" review document. Yes the TOB review is a point in time review, but so is this PR... or any assessment. I'm just trying to understand the impact/benefit of this piece of work when it feels like just about all of the info is already part of our repo (or should be as that's the source of truth for all things CE) so why duplicate it here, and are we then on the hook to keep it up to date since the minute the PR is merged it could be out of date?
> 
> Sorry if this sounds a bit like I'm pushing back, but I have a very strong negative reaction to things that feel like unnecessary bureaucracy - especially if it then requires an on-going commitment.

Sure, feel free to hop into a TAG Security meeting or to raise an issue there to discuss.  

I will say that your perspective has not been uncommon for folks at the beginning of the process.  I invite you to talk about folks who completed it to get their perspective.  Their candid feedback was that an assessment helped their project's security in a way that an audit did not.

duglin commented on assessments/projects/cloudevents/self-assessment.md:
```diff
@@ -0,0 +1,939 @@
+
+## Self-assessment use
+
+This self-assessment is created by the CloudEvents team to perform an internal
```
Can we modify this to more accurately state who created it? It's fair to add that the CE team has reviewed it but the CE team did not create it. There might be other spots in here that might need similar tweaking.

I'd also like to have something added about how the "source of truth" for all things CE related can be found in the CE project repos and this is a point-in-time summary of the project.

Igor8mr replied:
Thank you for the feedback on this. That makes sense, I changed it now to be more precise on who created and reviewed it.

> This security assessment was created by [Igor Rodrigues](https://github.com/Igor8mr), [Matthew Gong](https://github.com/MatthewZGong), [Kushal Kothari](https://github.com/Kushal-kothari) and [Devyani Bairagya](https://github.com/devyani-14) to perform an internal analysis of the CloudEvents project. The document was also reviewed by CloudEvents maintainers and members of the [CNCF Security Technical Advisory Group (TAG)](https://github.com/cncf/tag-security). It is meant to provide a current summary of the project and its security-related aspects. It is not intended to provide a security audit of CloudEvents, or function as an independent assessment or attestation of CloudEvents's security health. For the latest state of the project please check the [CloudEvents GitHub repository](https://github.com/cloudevents/spec).

I also added the paragraph below to the beginning of the document to mention where we extracted this information from:

> The information contained in this assessment was extracted from the [CloudEvents GitHub Repository](https://github.com/cloudevents/spec), and other channels maintained by the project, such as its [website](https://cloudevents.io/) and [Slack workspace](http://slack.cncf.io/).

Please, let me know if you want us to change these more or change other parts of the documents. Thank you for the help!

duglin commented on assessments/projects/cloudevents/self-assessment.md:
```diff
@@ -0,0 +1,939 @@
+## Self-assessment use
+
+This self-assessment is created by the CloudEvents team to perform an internal
+analysis of the project's security. It is not intended to provide a security
```
I'd change this to something like:
This assessment was created by ..., and has been reviewed by the CloudEvents team. It is meant to provide a point-in-time summary the state of the project and summarize its security related aspects. It is not intended to provide a security audit of CloudEvents, or function as an independent assessment or attestation of CloudEvents' security health. For the latest state of the project please see the CloudEvents github [repository](https://github.com/cloudevents/spec).

Igor8mr replied:
Thank you for the suggestion, I incorporated it into the assessment with some modifications. Please, let me know if you want to make more changes to it.

torinvdb commented on assessments/projects/cloudevents/self-assessment.md:
```diff
@@ -0,0 +1,951 @@
+# CloudEvents self-assessment
+
+November 26, 2023
+
```
```suggestion
# CloudEvents self-assessment

This assessment was created by community members as part of the [Security Pals](https://github.com/cncf/tag-security/issues/1102) process and is currently pending changes from the maintainer team.

November 26, 2023

```
@ragashreeshekar

JustinCappos reviewed (approved):
A solid Security Pals assessment!  LGTM
